src/core/manager.c

```diff
@@ -556,6 +558,46 @@ static int manager_default_environment(Manager *m) {
         return 0;
 }
 
+static const struct {
+        ExecDirectoryType directory;
+        uint64_t type;
+        const char *suffix;
+} paths_system[_EXEC_DIRECTORY_MAX] = {
+        { EXEC_DIRECTORY_RUNTIME, SD_PATH_SYSTEM_RUNTIME, NULL },
+        { EXEC_DIRECTORY_STATE, SD_PATH_SYSTEM_STATE_PRIVATE, NULL },
+        { EXEC_DIRECTORY_CACHE, SD_PATH_SYSTEM_STATE_CACHE, NULL },
+        { EXEC_DIRECTORY_LOGS, SD_PATH_SYSTEM_STATE_LOGS, NULL },
+        { EXEC_DIRECTORY_CONF, SD_PATH_SYSTEM_CONFIGURATION, NULL },
+}, paths_user[_EXEC_DIRECTORY_MAX] = {
+        { EXEC_DIRECTORY_RUNTIME, SD_PATH_USER_RUNTIME, NULL },
+        { EXEC_DIRECTORY_STATE, SD_PATH_USER_CONFIGURATION, NULL },
+        { EXEC_DIRECTORY_CACHE, SD_PATH_SYSTEM_STATE_CACHE, NULL },
+        { EXEC_DIRECTORY_LOGS, SD_PATH_SYSTEM_CONFIGURATION, "log" },
+        { EXEC_DIRECTORY_CONF, SD_PATH_SYSTEM_CONFIGURATION, NULL },
+};
```

Comment:
I'd define these static arrays inside of the manager_setup_prefix() function, we don't really need it outside.